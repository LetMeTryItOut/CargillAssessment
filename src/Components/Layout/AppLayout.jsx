import React from 'react';
import Header from '../Header/Header';

const AppLayout = (props) => {
    return (
        <div data-testid="applayout">
            <Header />
            <div>{props.children}</div>
        </div>
    );
}
export default AppLayout;