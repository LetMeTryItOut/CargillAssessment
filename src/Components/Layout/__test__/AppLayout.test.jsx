import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import AppLayout from '../AppLayout';

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(AppLayout, div);
});


it("renders AppLayout correctly", () => {
    const { getByTestId } = render(<AppLayout></AppLayout>);
    expect(getByTestId('applayout')).toHaveTextContent("Welcome to Countries Hub");
});


it("matches snapshot", () => {
    const component = renderer.create(<div style={{'text-align': 'center', 'font-weight': '900', 'backgroundColor': 'lightblue', 'height': '30px', 'marginTop': '-10px', 'paddingTop':'10px'}}>Welcome to Countries Hub</div>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
