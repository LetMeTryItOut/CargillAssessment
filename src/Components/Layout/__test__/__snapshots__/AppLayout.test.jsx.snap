// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightblue",
      "font-weight": "900",
      "height": "30px",
      "marginTop": "-10px",
      "paddingTop": "10px",
      "text-align": "center",
    }
  }
>
  Welcome to Countries Hub
</div>
`;
